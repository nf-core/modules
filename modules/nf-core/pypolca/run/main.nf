process PYPOLCA_RUN {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/pypolca:0.4.0--pyhdfd78af_0':
        'biocontainers/pypolca:0.4.0--pyhdfd78af_0' }"

    input:
    tuple val(meta) , path(reads)
    tuple val(meta2), path(contigs)

    output:
    tuple val(meta), path("${prefix}/*_corrected.fasta"), emit: polished
    tuple val(meta), path("${prefix}/*.vcf")            , emit: vcf
    tuple val(meta), path("${prefix}/*.report")         , emit: report
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args          = task.ext.args   ?: '--careful'
    prefix            = task.ext.prefix ?: "${meta.id}"
    def read_files    = reads instanceof List ? reads : [reads]
    def read_file_arg = read_files.size() > 1 ? "-1 ${read_files[0]} -2 ${read_files[1]}" : "-1 ${read_files[0]}"
    """
    gzip -cdf $contigs > contigs_uncompressed

    pypolca \
        run \
        -a contigs_uncompressed \\
        $read_file_arg \\
        -t ${task.cpus} \\
        -o ${prefix} \\
        --prefix ${prefix} \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pypolca: \$(pypolca --version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args   ?: ''
    prefix   = task.ext.prefix ?: "${meta.id}"
    """
    echo $args

    mkdir $prefix
    touch $prefix/${prefix}_corrected.fasta
    touch $prefix/${prefix}.vcf
    touch $prefix/${prefix}.report

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pypolca: \$(pypolca --version)
    END_VERSIONS
    """
}
