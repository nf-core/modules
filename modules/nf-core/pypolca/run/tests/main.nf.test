// nf-core modules test pypolca/run
nextflow_process {

    name "Test Process PYPOLCA_RUN"
    script "../main.nf"
    process "PYPOLCA_RUN"

    tag "modules"
    tag "modules_"
    tag "pypolca"
    tag "pypolca/run"
    tag "modules_nfcore"

    test("sarscov2 - fastq,fasta") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ], // reads
                ]
                input[1] = [
                    [ id:'meta2' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta.gz', checkIfExists: true), // assembly to polish
                ]
                """
            }
        }

        then {
            assertAll(
                { assert snapshot(
                    process.out.polished,
                    path(process.out.vcf[0][1]).vcf.variantsMD5,
                    process.out.report,
                    process.out.versions
                )}
            )
        }

    }

    test("sarscov2 - fastq,fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ], // reads
                ]
                input[1] = [
                    [ id:'meta2' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta.gz', checkIfExists: true), // assembly to polish
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
