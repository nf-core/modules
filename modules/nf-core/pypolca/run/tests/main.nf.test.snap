{
    "sarscov2 - fastq,fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_corrected.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.report:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,5be088b3b103d438d8621deab81f0aa3"
                ],
                "polished": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_corrected.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "report": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.report:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5be088b3b103d438d8621deab81f0aa3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-10-08T09:00:02.877616837"
    }
}